packages/storybook/cxl-ui/cxl-vaadin-accordion/cxl-playbook-accordion.story.js

```diff
@@ -2,18 +2,110 @@ import { html } from 'lit-html';
 import { unsafeHTML } from 'lit-html/directives/unsafe-html';
 import playbookStepData from './cxl-playbook-accordion.data.json';
 
+let tmpTimeoutId = 0;
+
+const hideAll = () => {
+  document.querySelectorAll(`[icon="vaadin:comment"]`).forEach((el) => {
+    el.classList.add('hidden');
+  });
+};
+
+const clearTimeout = () => {
+  window.clearTimeout(tmpTimeoutId);
+};
+
+const setDelay = () => {
+  clearTimeout();
+  tmpTimeoutId = window.setTimeout(() => hideAll(), 2000);
+};
+
+const isLayoutWide = () => document.querySelector('cxl-app-layout').wide;
+
+const onload = () => {
+  window.addEventListener('DOMContentLoaded', () => {
+    document.querySelectorAll('cxl-playbook-accordion').forEach((el) => {
+      el.addEventListener('mouseleave', () => {
+        setDelay();
+      });
+    });
+
+    document.querySelectorAll('vaadin-accordion-panel').forEach((el) => {
+      el.addEventListener('mouseenter', (evt) => {
+        if (!isLayoutWide()) {
+          return;
+        }
+        setDelay();
+
+        const stepId = evt.target.getAttribute('data-step-id');
+        const commentIcon = document.querySelector(
+          `[icon="vaadin:comment"][data-step-id="${stepId}"]`
+        );
+
+        hideAll();
+
+        if (commentIcon) {
+          commentIcon.classList.remove('hidden');
+        }
+      });
+
+      el.addEventListener('mouseleave', () => {
+        if (!isLayoutWide()) {
+          return;
+        }
+        setDelay();
+      });
+
+      el.querySelectorAll('iron-icon').forEach((icon) => {
+        icon.addEventListener('mouseenter', () => {
+          clearTimeout();
+        });
+
+        icon.addEventListener('mouseleave', () => {
+          setDelay();
+        });
+
+        icon.addEventListener('click', (evt) => {
+          // here should go the comment popup handler click
+          evt.stopImmediatePropagation();
+        });
+      });
+    });
+  });
```

Comment:
Can you clarify @dominik-stypula-polcode what all this dark margic is for? Completely unexpected to see so much nitty gritty JS involved :thinking:

Reply:
@lkraav nothing really complicated, this is just for the possibility to use the delayed icon hiding, so the icon does not suddenly disappear when a user hovers out of the panel element, and puts a cursor on top of the icon. Code would probably be a couple of times shorter if jQuery was involved. Could be probably also done another way if I would edit the template for the `vaadin-accordion-panel` (which is based on `vaadin-details` template really) and making a new component extending it (could be even more code in this case), and appending the icon after `<span part="toggle"></span>` in the template. So the fastest way for me was to position the icon with `absolute` and outside the panel - this also does not increase the panel`s width. Mostly I use `popover` plugins to do this.

Reply:
I don't get it: what's wrong with simple `:hover` CSS?

Reply:
> I don't get it: what's wrong with simple `:hover` CSS?

@dominik-stypula-polcode still need your input here.

Reply:
@lkraav maybe this example will show why exactly https://codepen.io/dominik-stypula-polcode/pen/poRmeXd

Reply:
> maybe this example will show why exactly

Because comment element is outside initial hover target, or? Be specific, we want to avoid incorrect translation risks.

If yes, solution is to have comment element inside hover element. Anything stopping such?

Reply:
`<slot name="summary"></slot>` goes before `<span part="toggle"></span>` which doesn't contain any slot in the vaadin details template, and the theme is **reverse**

Reply:
Going with simpler nested buttons for now.